airflow/models.py

```diff
@@ -1117,6 +1149,9 @@ def __init__(
         self.pool = pool
         self.sla = sla
         self.execution_timeout = execution_timeout
+        self.on_failure_callback = on_failure_callback
+        self.on_success_callback = on_success_callback
+        self.on_retry_callback = on_retry_callback
```

Comment:
oh... I meant, instead of passing a single callback for on_failure, pass a list, so you can do multiple things if a task fails without wrapping wrapping callbacks inside themselves.